frontend/packages/console-shared/src/components/popover-body/popover-body.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { DataPoint } from '@console/internal/components/graphs';
+import { ExternalLink, Humanize, resourcePathFromModel } from '@console/internal/components/utils';
+import { Dropdown } from '@console/internal/components/utils/dropdown';
+import { PodModel, NodeModel, ProjectModel } from '@console/internal/models';
+import { K8sKind } from '@console/internal/module/k8s';
+import { connectToURLs, MonitoringRoutes } from '@console/internal/reducers/monitoring';
+import { getPrometheusExpressionBrowserURL } from '@console/internal/components/graphs/prometheus-graph';
+import './popoverBody.scss';
+
+const dropdownKeys = ['By Pods', 'By Node', 'By Projects'];
+const models = [PodModel, NodeModel, ProjectModel];
+const dropdownItems = {
+  'By Pods': 'By Pods',
+  'By Node': 'By Node',
+  'By Projects': 'By Projects',
+};
+
+export const PopoverBody = connectToURLs(MonitoringRoutes.Prometheus)(
+  React.memo(
+    ({ topConsumers, title, humanize, query, urls }: React.PropsWithChildren<PopoverBodyProps>) => {
+      const [selectedFilter, setSelectedFilter] = React.useState(dropdownKeys[0]);
+      const [bodyData, setBodyData] = React.useState(null);
+
+      const popoverTitle = title ? `Top ${title} consumers` : `Top consumers`;
+      const selectedIndex = dropdownKeys.indexOf(selectedFilter);
+      const link = getPrometheusExpressionBrowserURL(urls, [`topk(20, ${query})`]);
```

Comment:
we are using `withDashboardResources` HOC for dashboard related queries

Reply:
also getting topk20 is not enough - we need to get all consumers and then filter the top5 which actually exists. See how its done in top consumers card https://github.com/openshift/console/blob/master/frontend/public/components/dashboard/dashboards-page/overview-dashboard/top-consumers-card.tsx#L140 which this popup should replace